sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/orc/OrcFileFormat.scala

```diff
@@ -154,11 +155,18 @@ class OrcFileFormat
 
     (file: PartitionedFile) => {
       val conf = broadcastedConf.value.value
+      val metaCacheEnabled =
+        conf.getBoolean(SQLConf.FILE_META_CACHE_ORC_ENABLED.key, false)
 
       val filePath = new Path(new URI(file.filePath))
 
       val fs = filePath.getFileSystem(conf)
-      val readerOptions = OrcFile.readerOptions(conf).filesystem(fs)
+      val readerOptions = if (metaCacheEnabled) {
+        val tail = OrcFileMeta.readTailFromCache(filePath, conf)
+        OrcFile.readerOptions(conf).filesystem(fs).orcTail(tail)
```

Comment:
What happen if the file is replaced?

Reply:
This is a very good problem. If we want to handle it well, we need a relatively more complex design, such as adding a command to trigger the relevant cleaning of all executors when the file changes.

However, if the file changes are not perceived by spark, I think wrong data will be read here

Reply:
If the file name has the timestamp, I think we don't have to worry too much. The names of the new file and the old file are different and  they can ensure that they don't read the wrong data.

Reply:
If it is manually replaced and the file has the same name and the corresponding file meta exists in the cache, an incorrect file meta will be used to read the data. If the data reading fails, the job will fail. But if the data reading happens to be successful, the job will read the wrong data.

Reply:
In fact, even if there is no `FileMetaCache`, there is a similar risk in manually replace files with same name, because the offset and length  of `PartitionedFile` maybe don't match after manually replace.